File: ChatBot/ChatBot/Message.m
#import "Message.h"

@implementation Message

+ (instancetype)messageWithContent:(NSString *)content isUser:(BOOL)isUser {
    Message *message = [[Message alloc] init];
    message.content = content;
    message.isUser = isUser;
    message.timestamp = [NSDate date];
    return message;
}

@end 

File: ChatBot/ChatBot/Message.swift
import Foundation

class Message {
    let content: String
    let isUser: Bool
    var timestamp: Date
    
    init(content: String, isUser: Bool) {
        self.content = content
        self.isUser = isUser
        self.timestamp = Date()
    }
    
    static func message(content: String, isUser: Bool) -> Message {
        return Message(content: content, isUser: isUser)
    }
} 
